fpm/src/fpm_targets.f90

```diff
@@ -409,7 +409,7 @@ function find_module_dependency(targets,module_name,include_dir) result(target_p
                     exit
                 case default
                     if (present(include_dir)) then
-                        if (dirname(targets(k)%ptr%source%file_name) == include_dir) then
+                        if (index(dirname(targets(k)%ptr%source%file_name), include_dir) > 0) then ! source file is within the include_dir or a subdirectory
```

Comment:
Should we check if index is 1 rather than larger zero? A match at position 10 could be a false positive.

Reply:
I suppose that's true. Good catch.